### DISCUSSION TITLE
UI ideas

### DISCUSSION THREAD

**Discussion by benthillerkus** [DSchroer/dslcad#7] [Ideas]
It'd be nice if each variable that is assigned with a number would get a small slider in the viewport for tweaking.
Also, assigning labels to points and edges would be useful. Maybe if after the declaration of such a thing a `// comment` follows, that comment could then be shown in the Viewport next to the element.

**Comment by DSchroer** [comment]
So this hasnt been fleshed out yet but I have started laying the groundwork for this feature already. Essentially all variables are also parameters:
```
// my_script.ds

var t; // required parameter
var i = 5; // optional parameter
```

You can call any of those from another script like so:
```
./my_script(t=10, i=50);
```

So the ideal would be to simply expose the top level variables when running a script in a window. Would that work for you? 

There are two differences that I see between my plan and yours.
1. You have typed values since a slider would only ever work for numbers (maybe annotations like `var t: number;` would work)
2. We probably don't want all variables to be editable (maybe making it start with underscore `var _i` would make it private)

What do you think?

**Reply by benthillerkus** [reply]
I feel like if it was textboxes, there wouldn't be much benefit over just editing the values in the script.

Re which parameters to show, I think having different declarations or modifiers would probably be better, something like `pub b;` or `pub var b;` or `parameter b`; or `tweak b;`.
The problem with a `_` convention is that it makes it harder to switch from either into the other one

**Reply by DSchroer** [reply]
> I feel like if it was textboxes

I'm not sure. We can always start with that and progress to sliders and such. It expands out nicely from a development point of view.

**Reply by DSchroer** [reply]
I think for now `var b;` is clear that it should be editable since it is variable. I like the quick ability to change it so maybe `let b;` would make it private.

**Reply by DSchroer** [reply]
Looks like there is another proposal that could solve the private fields issue. What do you think of this?
https://github.com/DSchroer/dslcad/issues/13
